test/ionOverlayMethodCreateShouldUseAwaitRule.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { ruleMessage, ruleName } from '../src/ionOverlayMethodCreateShouldUseAwaitRule';
+import { assertAnnotated, assertSuccess } from './testHelper';
+
+describe(ruleName, () => {
+  describe('success', () => {
+    it('should work when await exists in front of the create method', () => {
+      let source = `
+      class DoSomething{
+        constructor(private actionSheetController: ActionSheetController){}
+        
+        doWork(){
+          await this.actionSheetController.create({
+            component: PopoverComponent,
+            ev: event,
+            translucent: true
+          })
+        }
+      }
+        `;
+      assertSuccess(ruleName, source);
+    });
+
+    it('should not do anything if the class name does not match', () => {
+      let source = `
+      class DoSomething{
+        constructor(private someOtherController: SomeOtherController){}
+        
+        doWork(){
+          await this.someOtherController.create({
+            component: PopoverComponent,
+            ev: event,
+            translucent: true
+          })
+        }
+      }
+        `;
+      assertSuccess(ruleName, source);
+    });
+  });
+
+  const controllersList = [
+    'ActionSheetController',
+    'AlertController',
+    'LoadingController',
+    'ModalController',
+    'PopoverController',
+    'ToastController'
+  ];
+
+  for (let controller of controllersList) {
+    describe(controller, () => {
+      describe('failure', () => {
+        it('should fail if await is not present', () => {
+          let source = `
+      class DoSomething{
+        constructor(private actionSheetController: ${controller}){}
```

Comment:
We may want to test for when the property is named something other than `this.actionSheetController`